src/chunk_append/explain.c

```diff
@@ -0,0 +1,163 @@
+									 NULL,
+									 list_length(state->initial_subplans) -
+										 list_length(node->custom_ps),
+									 es);
+
+	if (state->runtime_exclusion)
+		ExplainPropertyIntegerCompat("Chunks excluded during runtime",
+									 NULL,
+									 list_length(state->filtered_subplans) -
+										 bms_num_members(state->valid_subplans),
+									 es);
+}
+
+/*
+ * adjusted from postgresql explain.c
+ * since we have to keep the state in custom_private our sort state
+ * is in lists instead of arrays
+ */
+static void
+show_sort_group_keys(ChunkAppendState *state, List *ancestors, ExplainState *es)
+{
+	Plan *plan = state->csstate.ss.ps.plan;
+	List *context;
+	List *result = NIL;
+	StringInfoData sortkeybuf;
+	bool useprefix;
+	int keyno;
+	int nkeys = list_length(linitial(state->sort_options));
+	List *sort_indexes = linitial(state->sort_options);
+	List *sort_ops = lsecond(state->sort_options);
+	List *sort_collations = lthird(state->sort_options);
+	List *sort_nulls = lfourth(state->sort_options);
+
+	if (nkeys <= 0)
+		return;
+
+	initStringInfo(&sortkeybuf);
+
+	/* Set up deparsing context */
+	context = set_deparse_context_planstate(es->deparse_cxt, (Node *) state, ancestors);
+	useprefix = (list_length(es->rtable) > 1 || es->verbose);
+
+	for (keyno = 0; keyno < nkeys; keyno++)
+	{
+		/* find key expression in tlist */
+		AttrNumber keyresno = list_nth_oid(sort_indexes, keyno);
+		TargetEntry *target = get_tle_by_resno(plan->targetlist, keyresno);
+		char *exprstr;
+
+		if (!target)
+			elog(ERROR, "no tlist entry for key %d", keyresno);
+		/* Deparse the expression, showing any top-level cast */
+		exprstr = deparse_expression((Node *) target->expr, context, useprefix, true);
+		resetStringInfo(&sortkeybuf);
+		appendStringInfoString(&sortkeybuf, exprstr);
+		/* Append sort order information, if relevant */
+		if (sort_ops != NIL)
+			show_sortorder_options(&sortkeybuf,
+								   (Node *) target->expr,
+								   list_nth_oid(sort_ops, keyno),
+								   list_nth_oid(sort_collations, keyno),
+								   list_nth_oid(sort_nulls, keyno));
+		/* Emit one property-list item per sort key */
+		result = lappend(result, pstrdup(sortkeybuf.data));
+	}
+
+	ExplainPropertyList("Order", result, es);
+}
+
+/* copied verbatim from postgresql explain.c */
+static void
+show_sortorder_options(StringInfo buf, Node *sortexpr, Oid sortOperator, Oid collation,
+					   bool nullsFirst)
+{
+	Oid sortcoltype = exprType(sortexpr);
+	bool reverse = false;
+	TypeCacheEntry *typentry;
+
+	typentry = lookup_type_cache(sortcoltype, TYPECACHE_LT_OPR | TYPECACHE_GT_OPR);
+
+	/*
+	 * Print COLLATE if it's not default.  There are some cases where this is
+	 * redundant, eg if expression is a column whose declared collation is
+	 * that collation, but it's hard to distinguish that here.
+	 */
+	if (OidIsValid(collation) && collation != DEFAULT_COLLATION_OID)
+	{
+		char *collname = get_collation_name(collation);
```

Comment:
This whole branch is never tested. Please test non-default collations

Reply:
can't be tested in our ci because alpine does not support locales